src/NServiceBus.AcceptanceTests/Hosting/When_overriding_local_addresses.cs

```diff
@@ -0,0 +1,66 @@
+namespace NServiceBus.AcceptanceTests.Hosting
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+    using NUnit.Framework;
+
+    public class When_overriding_local_addresses : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_use_custom_queue_names()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<MyEndpoint>(e => e.When(b => b.SendLocal(new MyMessage())))
+                .Done(c => c.Done)
+                .Run();
+
+            Assert.IsTrue(context.Done);
+            Assert.IsTrue(context.InputQueue.StartsWith("OverriddenLocalAddress"));
+        }
+
+        public class MyEndpoint : EndpointConfigurationBuilder
+        {
+            public MyEndpoint()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.EnableFeature<TimeoutManager>();
+                    var transportExtensions = new TransportExtensions(c.GetSettings());
+                    transportExtensions.AddAddressTranslationRule(address =>
+                    {
+                        return "OverriddenLocalAddress" + address.Qualifier; //Overriding -> Overridden
+                    });
+                });
+            }
+        }
+
+        public class MyMessageHandler : IHandleMessages<MyMessage>
+        {
+            public Context Context { get; set; }
+
+            public Task Handle(MyMessage message, IMessageHandlerContext context)
+            {
+                Context.Done = true;
```

Comment:
Nitpick: Done should be assigned last since the test might stop before .InputQueue has been assigned